workflow/controller/controller.go

```diff
@@ -1054,8 +1057,8 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 						log.WithError(err).Error("Failed to list pods")
 					}
 					for _, p := range podList.Items {
-						if err := wfc.enablePodForDeletion(ctx, pods, p.Namespace, p.Name, false); err != nil {
-							log.WithError(err).Error("Failed to enable pod for deletion")
+						if slice.ContainsString(p.Finalizers, common.FinalizerPodStatus) {
+							wfc.queuePodForCleanup(p.Namespace, p.Name, removeFinalizer)
```

Comment:
would `deletePod` not make sense here? as opposed to creating a new `removeFinalizer` action?

Reply:
It is cased by this case: https://github.com/argoproj/argo-workflows/blob/f0a867b6325ec3f20369bf6c92518dc5bb2ebf59/test/e2e/pod_cleanup_test.go#L120
Workflow get deleted, so pod get deleted, too, so waiting pod completed wouldn't happen, but I have not found who deleted it and when. About the failure in Windows Tests, I am testing it in a virtual machine, but, i am struggling with network problems(see my location...). I need some help.

Reply:
You can ignore the Windows tests, they're not required to pass as they're still unstable